### PR TITLE
feat: support passing options to graphql validate and parse

### PR DESCRIPTION
**Pull request by markrzen**
Support passing options to GraphQL's `validate` and `parse` functions.

Reasoning to allow configuration of these functions topics is fine tuning to prevent DDOS attacks.

This allows developers to set:
1. `validate`'s options of `maxErrors` to something more/less than the 100 default. This prevents an attacker from specifying 100 invalid fields, requiring CPU time generating validation errors.
2. `parse`'s options of `maxTokens` to a value. It has no default. This prevent an attacker from producing a overly high complexity query that requires a high amount of CPU time.

### CODE REVIEW

**Comment by markrzen** [comment]
@mcollina Can we get this on a reviewers queue? I wouldn't usually poke maintainers like this, but previous PRs seem to be getting reviewed within a few days. I wanted to make sure this didn't fall through the cracks.

**Comment by markrzen** [comment]
@simoneb Any chance this is something you can review?

File: docs/api/options.md
Comment (simoneb) [inline docs/api/options.md:39]: I would avoid deep links to source code, as it can change any time. The alternative would be to use permalinks, but I'm not sure that's a good option either. Is there official documentation that we can point to instead?

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js
Comment (simoneb) [inline index.js:439]: relying on an external library's error message to decide what to do doesn't feel very reliable

Comment (markrzen) [reply index.js:439]: Totally agreed. Sadly I could not find another option. Any suggestion?

I did protect against it's potential change in the unit tests below.

File: docs/api/options.md
Comment (markrzen) [reply docs/api/options.md:39]: The published graphql docs do not contain option properties sadly. I could update them to a specific tag if that works for you?

Comment (simoneb) [reply docs/api/options.md:39]: not many compelling options then. I'm not sure. if we point at master, those links may refer to wrong lines when those files change. if we use permalinks, the reference will keep existing but the real code might have changed. not sure what's best really. shall we link to the files without any line number perhaps? that would give us some confidence that those files will keep existing and leave it to the reader to figure out where to find the types

File: index.js
Comment (simoneb) [reply index.js:439]: I guess we don't have many options here either. What will happen is that when dependabot bumps this dependency, the unit test will break which will give us some clue of what needs fixing.

File: docs/api/options.md
Comment (markrzen) [reply docs/api/options.md:39]: We could just enumerate the literal properties, rather than linking off. It does mean our future selfs will need to go back to the docs and confirm when things have been added/removed.

Not sure there is a good option, other than pushing for these functions to be documented more fully in GraphQL library.

Happy to change to without line numbers if that makes anyone feel more comfortable.

Comment (simoneb) [reply docs/api/options.md:39]: No line numbers is the compromise I'd go with, but happy for somebody else to chime in too

Comment (markrzen) [reply docs/api/options.md]: Done! :)

**Review by simoneb — APPROVED** [review]
LGTM, thanks for this!

**Comment by markrzen** [comment]
@simoneb @mcollina Thanks y'all. What sort of release schedule are y'all doing? Any expectations I should have?

**Pull request merged**
